corehq/apps/accounting/interface.py

```diff
@@ -406,15 +406,176 @@ def get_subtotal_and_deduction(line_items):
+                    invoice.account.name
+                ),
+                invoice.get_domain(),
+                "YES" if new_this_month else "no",
+                contact_info.company_name,
+                contact_info.emails,
+                contact_info.first_name,
+                contact_info.last_name,
+                contact_info.phone_number,
+                contact_info.first_line,
+                contact_info.second_line,
+                contact_info.city,
+                contact_info.state_province_region,
+                contact_info.postal_code,
+                contact_info.country,
+                invoice.date_start.strftime(USER_DATE_FORMAT),
+                invoice.date_end.strftime(USER_DATE_FORMAT),
+                invoice.date_due.strftime(USER_DATE_FORMAT),
+                get_exportable_column(invoice.subtotal),
+                get_exportable_column(invoice.balance),
+                "Paid" if invoice.is_paid else "Not paid",
+                "YES" if invoice.is_hidden else "no",
+            ]
+
+            invoice_url = reverse(WireInvoiceSummaryView.urlname, args=(invoice.id,))
+            if not self.is_rendered_as_email:
+                columns.extend([
+                    mark_safe(make_anchor_tag(invoice_url, 'Go to Invoice'))
+                ])
+            rows.append(columns)
+        return rows
+
+    @property
+    @memoized
+    def filters(self):
+        filters = {}
+
+        domain_name = DomainFilter.get_value(self.request, self.domain)
+        if domain_name is not None:
+            filters.update(domain=domain_name)
+
+        payment_status = \
+            PaymentStatusFilter.get_value(self.request, self.domain)
+        if payment_status is not None:
+            filters.update(
+                date_paid__isnull=(
+                    payment_status == PaymentStatusFilter.NOT_PAID
+                ),
+            )
+
+        statement_period = \
+            StatementPeriodFilter.get_value(self.request, self.domain)
+        if statement_period is not None:
+            filters.update(
+                date_start__gte=statement_period[0],
+                date_start__lte=statement_period[1],
+            )
+
+        due_date_period = \
+            DueDatePeriodFilter.get_value(self.request, self.domain)
+        if due_date_period is not None:
+            filters.update(
+                date_due__gte=due_date_period[0],
+                date_due__lte=due_date_period[1],
+            )
+
+        is_hidden = IsHiddenFilter.get_value(self.request, self.domain)
+        if is_hidden is not None:
+            filters.update(
+                is_hidden=(is_hidden == IsHiddenFilter.IS_HIDDEN),
+            )
+
+        filters.update(is_hidden_to_ops=False)
+
+        return filters
+
+    @property
+    @memoized
+    def invoices(self):
+        return WireInvoice.objects.filter(**self.filters)
```

Comment:
FYI chaining filters is more common than building and passing a ```dict``` (contrary to other examples in this file) (no fix required)